packages/hardhat-core/src/internal/sentry/anonymizer.ts

```diff
@@ -226,4 +234,80 @@ export class Anonymizer {
 
     return result;
   }
+
+  private _anonymizeMnemonic(
+    errorMessage: string,
+    mnemonicPhraseLengthThreshold: number,
+    minimumAmountOfWordsToAnonymize: number
+  ): string {
+    const fragments = errorMessage.split(/[!$%^&*()_+|~\-=`{}[\]:;<>?"',.]/gu);
+    const fragmentsToBeAnonymized: string[] = [];
+    // If there are 10 consecutive words, there's a good chance of there being a mnemonic phrase
+    const wordlist = ([] as string[]).concat(
+      ...[
+        require("ethereum-cryptography/bip39/wordlists/czech"),
+        require("ethereum-cryptography/bip39/wordlists/english"),
+        require("ethereum-cryptography/bip39/wordlists/french"),
+        require("ethereum-cryptography/bip39/wordlists/italian"),
+        require("ethereum-cryptography/bip39/wordlists/japanese"),
+        require("ethereum-cryptography/bip39/wordlists/korean"),
+        require("ethereum-cryptography/bip39/wordlists/simplified-chinese"),
+        require("ethereum-cryptography/bip39/wordlists/spanish"),
+        require("ethereum-cryptography/bip39/wordlists/traditional-chinese"),
+      ].map((wordlistModule) => wordlistModule.wordlist)
+    );
+    for (const [fragmentIndex, fragment] of fragments.entries()) {
+      // TODO: split on whitespace instead of spaces?
+      const words = fragment.split(" ");
```

Comment:
I think there's no downside on using `/\s+/` here.

Reply:
Besides, this means that the heuristic won't work if the mnemonic is separated by line breaks, right? Not sure how likely that is, but given what's at stake I'd rather be over-paranoid.

Reply:
That's right. The problem is that I need to reconstruct the string as it originally was. I think `String.prototype.split` doesn't help much here...

Reply:
Actually, now that I think about it, it might be fine to mangle the string a bit just to cover these additional cases.

Reply:
Yes, I'm afraid doing this requires some careful usage of regex. I'm still inclined to do it, but I'm not 100% sure. @alcuadrado wdyt?

Reply:
Maybe split with `/\b/`, and then make the algorithm ignore the strings that match `/\s+/`?

Reply:
`\b` is word boundary, so for example

```js
"asd asd asd".split(/\b/)
```

is

```js
[ 'asd', ' ', 'asd', ' ', 'asd' ]
```

Reply:
Sadly, it seems it's not enough:
```js
> '就 就 就 就 就 就 就 就 就 就 就 就'.split(/\b/u)
[ '就 就 就 就 就 就 就 就 就 就 就 就' ]
```

Reply:
I forgot to update this here, but we talked about using `/\p{Letter}+/u` to get the list of words of a string. It seems robust enough.

Reply:
I'd split it manually rather than using something that obscure. Something like this:

```js
function splitByWordBoundaries(str) {
  const parts = [];
  let withinWord = false;

  let current = "";
  for (const codePoint of str) {
    if (codePoint.match(/\s/u)) {
      if (withinWord) {
        if (current !== "") {
          parts.push(current);
        }
        current = codePoint;
        withinWord = false;
      } else {
        current += codePoint;
      }
    } else {
      if (withinWord) {
        current += codePoint;
      } else {
        if (current !== "") {
          parts.push(current);
        }
        current = codePoint;
        withinWord = true;
      }
    }
  }

  if (current !== "") {
    parts.push(current);
  }

  return parts;
}
```